core/rid.h

```diff
@@ -92,63 +101,41 @@ class RID_OwnerBase {
 		p_rid._data = p_data;
 		refcount.ref();
 		p_data->_id = refcount.get();
-#ifndef DEBUG_ENABLED
 		p_data->_owner = this;
-#endif
 	}
 
-#ifndef DEBUG_ENABLED
+	_FORCE_INLINE_ void _free(RID &p_rid) {
+		p_rid._data = nullptr;
+	}
 
 	_FORCE_INLINE_ bool _is_owner(const RID &p_rid) const {
 		return this == p_rid._data->_owner;
 	}
 
 	_FORCE_INLINE_ void _remove_owner(RID &p_rid) {
-		p_rid._data->_owner = NULL;
+		p_rid._data->_owner = nullptr;
 	}
-#endif
 
 public:
 	virtual void get_owned_list(List<RID> *p_owned) = 0;
-
 	static void init_rid();
 	virtual ~RID_OwnerBase() {}
 };
 
 template <class T>
 class RID_Owner : public RID_OwnerBase {
-public:
-#ifdef DEBUG_ENABLED
-	mutable Set<RID_Data *> id_map;
-#endif
 public:
 	_FORCE_INLINE_ RID make_rid(T *p_data) {
 		RID rid;
 		_set_data(rid, p_data);
-
-#ifdef DEBUG_ENABLED
-		id_map.insert(p_data);
-#endif
-
```

Comment:
If you get rid of the `id_map`, does the RID_Owner system still do anything? i.e. Is it just a debug system for keeping RIDs together and checking, but if these are all pass throughs in your PR it becomes redundant?